x-pack/metricbeat/module/iis/application_pool/reader.go

```diff
@@ -44,21 +44,40 @@ type WorkerProcess struct {
 }
 
 var appPoolCounters = map[string]string{
-	"process.pid":                          "\\Process(w3wp*)\\ID Process",
-	"process.cpu_usage_perc":               "\\Process(w3wp*)\\% Processor Time",
-	"process.handle_count":                 "\\Process(w3wp*)\\Handle Count",
-	"process.thread_count":                 "\\Process(w3wp*)\\Thread Count",
-	"process.working_set":                  "\\Process(w3wp*)\\Working Set",
-	"process.private_bytes":                "\\Process(w3wp*)\\Private Bytes",
-	"process.virtual_bytes":                "\\Process(w3wp*)\\Virtual Bytes",
-	"process.page_faults_per_sec":          "\\Process(w3wp*)\\Page Faults/sec",
-	"process.io_read_operations_per_sec":   "\\Process(w3wp*)\\IO Read Operations/sec",
-	"process.io_write_operations_per_sec":  "\\Process(w3wp*)\\IO Write Operations/sec",
+	"process.pid":                         "\\Process(w3wp*)\\ID Process",
+	"process.cpu_usage_perc":              "\\Process(w3wp*)\\% Processor Time",
+	"process.handle_count":                "\\Process(w3wp*)\\Handle Count",
+	"process.thread_count":                "\\Process(w3wp*)\\Thread Count",
+	"process.working_set":                 "\\Process(w3wp*)\\Working Set",
+	"process.private_bytes":               "\\Process(w3wp*)\\Private Bytes",
+	"process.virtual_bytes":               "\\Process(w3wp*)\\Virtual Bytes",
+	"process.page_faults_per_sec":         "\\Process(w3wp*)\\Page Faults/sec",
+	"process.io_read_operations_per_sec":  "\\Process(w3wp*)\\IO Read Operations/sec",
+	"process.io_write_operations_per_sec": "\\Process(w3wp*)\\IO Write Operations/sec",
+
+	// .NET CLR Memory
+	"net_clr.bytes_in_all_heaps":      "\\.NET CLR Memory(w3wp*)\\# Bytes in all Heaps",
```

Comment:
Can the fields grouped based on category. Ex: net_clr.**memory**. bytes_in_all_heaps? What I observe is that the exception fields which exists are not grouped. If we group fields then we can do it for memory and locksandthreads only. WDYT?

Reply:
done